### PR TITLE
SLING-10147 block unauthorized access to SlingBindingsVariablesListJsonServlet

### PR DESCRIPTION
**Pull request by enapps-enorman**
Require WebConsoleSecurityProvider2 service to exist and WebConsoleSecurityProvider2#authenticate to return true before allowing access

This is alternative way to solve [SLING-10147](https://issues.apache.org/jira/browse/SLING-10147) with less code than #5

### CODE REVIEW

File: src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java [hunk 1]
```diff
@@ -91,6 +102,24 @@ protected void activate(ComponentContext context) {
     @Override
     protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response)
             throws ServletException, IOException {
+        if (webconsoleSecurity == null) {
+            log("Acccess forbidden as the WebConsoleSecurity reference is not set");
```
Comment (kwin) [inline src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java:106]: small typo: Acccess -> access

File: src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java [hunk 2]
```diff
@@ -91,6 +102,24 @@ protected void activate(ComponentContext context) {
...
+            response.sendError(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        } else if (!(webconsoleSecurity instanceof WebConsoleSecurityProvider2)) {
+            log("Acccess forbidden as the WebConsoleSecurity reference does not implement WebConsoleSecurityProvider2");
```
Comment (kwin) [inline src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java:110]: small typo: Acccess -> access

File: src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java [hunk 3]
```diff
@@ -91,6 +102,24 @@ protected void activate(ComponentContext context) {
...
+            response.sendError(HttpServletResponse.SC_FORBIDDEN);
+            return;
+        } else if (!((WebConsoleSecurityProvider2)webconsoleSecurity).authenticate(request, response)) {
+            // the request is terminated without any more response sent back to the client.
+            //    The WebConsoleSecurityProvider2 implementation may have sent auth challenge to the client
+            //    in the case of anonymous access.
+            if (!response.isCommitted()) {
```
Comment (kwin) [inline src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java:117]: why the check for isCommitted only here and not for the other two cases?

File: src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java (hunk 1 above)
Comment (enapps-enorman) [reply src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java:106]: whoops.  I'll fix that now.

File: src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java (hunk 3 above)
Comment (enapps-enorman) [reply src/main/java/org/apache/sling/scripting/core/impl/SlingBindingsVariablesListJsonServlet.java:120]: I guess I didn't think there was a code path where the response was touched before the servlet was called.  But I suppose it is possible that some filter has done something before calling the servlet, so I'll change it to do the same check for all the cases.

File: pom.xml [hunk 4]
```diff
@@ -376,5 +383,11 @@
             <version>1.2.6</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.sling</groupId>
+            <artifactId>org.apache.sling.extensions.webconsolesecurityprovider</artifactId>
+            <version>1.2.4</version>
+            <scope>test</scope>
```
Comment (rombert) [inline pom.xml:390]: Apologies for the late review @enapps-enorman , but is the `test` scope the right one here?

Comment (kwin) [reply pom.xml:390]: IMHO yes, as the API is part of Felix. The Sling Web Console Security Provider is only one implementation which is used in ITs.

Comment (rombert) [reply pom.xml:390]: Ah, got it. Thanks @kwin

**Pull request merged**